Docs: Fix broken links in plugin-types-usage

jgomezselles opened the pull request:
<!--

Thank you for sending a pull request! Here are some tips:

1. If this is your first time, please read our contribution guide at https://github.com/grafana/plugin-tools/blob/main/CONTRIBUTING.md

2. Ensure you include and run the appropriate tests as part of your Pull Request.

3. In a new feature or configuration option, an update to the documentation is necessary. Everything related to the documentation is under the docs folder in the root of the repository.

4. If the Pull Request is a work in progress, make use of GitHub's "Draft PR" feature and mark it as such.

5. If you can not merge your Pull Request due to a merge conflict, Rebase it. This gets it in sync with the main branch.

6. Name your PR as "<FeatureArea>: Describe your change", e.g. Alerting: Prevent race condition. If it's a fix or feature relevant for the changelog describe the user impact in the title. The PR title is used to auto-generate the changelog for issues marked with the "add to changelog" label.

-->

**What this PR does / why we need it**:
This PR fixes some links which were not working in [plugin-types-usage](https://grafana.com/developers/plugin-tools/key-concepts/plugin-types-usage) page.

**Which issue(s) this PR fixes**:
None that I could find.
<!--

- Automatically closes linked issue when the Pull Request is merged.

Usage: "Fixes #<issue number>", or "Fixes (paste link of issue)"

-->

Fixes #

**Special notes for your reviewer**:
It feels like full links are mixed with references and there could be a better/more unified way of fixing links, however, this PR should serve as a quick fix to, at least, get those links up again!

jgomezselles committed:
Fix broken links in plugin-types-usage

Some links were not working, so this commit fixes it by
correcting the routes.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/grafana/plugin-tools?pullRequest=1056) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/grafana/plugin-tools?pullRequest=1056) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/grafana/plugin-tools?pullRequest=1056) it.</sub>

academo commented on docusaurus/docs/key-concepts/plugin-types-usage.md:
```diff
@@ -30,7 +30,7 @@ Grafana plugin development allows for many options depending on the type of user
 - **App plugin** - an integrated out-of-the-box experience.
 
-Refer to [Get started](../get-started/get-started.mdx) for instructions on how to quickly scaffold [each type](https://grafana.com/developers/plugin-tools/reference/prompts.md/#what-type-of-plugin-would-you-like) of plugin.
+Refer to [Get started](../get-started/get-started.mdx) for instructions on how to quickly scaffold [each type](https://grafana.com/developers/plugin-tools/reference/prompts#select-a-plugin-type) of plugin.
```
Thanks for spotting these issues. We should replace this (and all the other links in this PR) with relative links instead of these absolute ones. Could you please update your PR with those?

jgomezselles replied:
Thanks for the review and suggestion, @academo ! Will do!

jgomezselles committed:
Docs: Use relative links

Previous fix was using complete url instead of relative links.
This commit fixes that.

academo committed:
Merge branch 'main' into docs-fix-links-plugin-types-usage
